plugins/woocommerce/includes/admin/helper/class-wc-helper.php

```diff
@@ -681,6 +683,31 @@ public static function current_screen( $screen ) {
 		}
 	}
 
+	private static function maybe_redirect_to_new_marketplace_installer() {
+		// Redirect requires the "install" URL parameter to be passed.
+		if ( empty( $_GET[ 'install' ] ) ) {
+			return;
+		}
+
+		// Check the format of the product key in the "install" parameter is what we expect it to be
+		if ( ! preg_match( '/^W00-[a-f0-9]{8}-[a-f0-9]{4}-[a-f0-9]{4}-[a-f0-9]{12}$/i', $_GET[ 'install' ], $matches ) ) {
```

Comment:
Let’s remove this check. The structure is different for promo products (WooPayments) and it may change in the future, which would break installs.

Reply:
Done in https://github.com/woocommerce/woocommerce/pull/42010/commits/2b97ca63d7d36e228c1d6ca4bc34983ea54fdb6c